Application/EdFi.Ods.Api/Helpers/AssemblyLoaderHelper.cs

```diff
@@ -146,6 +147,52 @@ public static IEnumerable<string> FindPluginAssemblies(string pluginFolder, bool
                 yield break;
             }
 
+            var apiModelFiles = Directory.GetFiles(pluginFolder, "ApiModel-EXTENSION.json", SearchOption.AllDirectories);
+
+            var physicalNames = new List<KeyValuePair<string, string>>();
+
+            Array.Sort(apiModelFiles, StringComparer.InvariantCulture);
+
+            foreach (var apiModelFilePath in apiModelFiles)
+            {
+                var myJsonString = File.ReadAllText(apiModelFilePath);
+
+                var myJsonObject = JObject.Parse(myJsonString);
+
+                JToken physicalNameJToken = myJsonObject.SelectToken("$.schemaDefinition.physicalName");
+
+                if (physicalNameJToken != null)
+                {
+                    string[] folderNames = apiModelFilePath.Split(Path.DirectorySeparatorChar);
+                    foreach (string folder in folderNames)
+                    {
+                        if (folder.Contains("Extensions"))
+                        {
+                            string key = physicalNameJToken.Value<string>();
+                            var element = new KeyValuePair<string, string>(key, folder);
+                            physicalNames.Add(element);
+                        }
+                    }
+                } 
+            }
+
+            bool isDuplicate = false;
+            var duplicateExtensionPlugins = physicalNames.ToLookup( x => x.Key).Where(x => x.Count() > 1);
+
+            foreach (var duplicateExtensionPlugin in duplicateExtensionPlugins)
+            {
+                isDuplicate = true;
+                var pluginFolders = duplicateExtensionPlugin.Select(s => s.Value); 
+                _logger.Error($"found duplicate extension schema name '{duplicateExtensionPlugin.Key}' in plugin folder." +
+                    $" You will be able to deploy only one of the following plugins '{string.Join("' and '", pluginFolders)}' folder name." +
+                    $" Please remove the conflicting plugins and retry");
+            }

```

Comment:
After latest changes I'm getting incorrect error messages again:
- found duplicate extension schema name 'homograph' in plugin folder. You will be able to deploy only one of the following plugins 'EdFi.Suite3.Ods.Extensions.Homograph.5.3.11 copy' and 'EdFi.Suite3.Ods.Extensions.Homograph.5.3.11' folder name. Please remove the conflicting plugins and retry
- found duplicate extension schema name 'sample' in plugin folder. You will be able to deploy only one of the following plugins `'EdFi.Suite3.Ods.Extensions.Homograph.5.3.11' and 'EdFi.Suite3.Ods.Extensions.Sample.5.3.10' and 'EdFi.Suite3.Ods.Extensions.Sample.5.3.10 copy' and 'EdFi.Suite3.Ods.Extensions.Sample.5.3.10'` folder name. Please remove the conflicting plugins and retry
- found duplicate extension schema name 'tpdm' in plugin folder. You will be able to deploy only one of the following plugins `'EdFi.Suite3.Ods.Extensions.Homograph.5.3.11' and 'EdFi.Suite3.Ods.Extensions.TPDM.Core.1.1.0.5.3.17' and 'EdFi.Suite3.Ods.Extensions.TPDM.Core.1.1.0.5.3.17 copy' and 'EdFi.Suite3.Ods.Extensions.TPDM.Core.1.1.0.5.3.17'` folder name. Please remove the conflicting plugins and retry


I think part of the problem is that the `physicalNames` list contains incorrect key/value pairs and duplicates itself:
![image](https://user-images.githubusercontent.com/23484186/134699811-5764c689-1889-4663-9635-5071c5d86229.png)